extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/SmallRyeReactiveMessagingProcessor.java

```diff
@@ -166,8 +205,268 @@ public void build(SmallRyeReactiveMessagingRecorder recorder, BeanContainerBuild
+                + HashUtil.sha1(sigBuilder.toString());
+
+        try (ClassCreator invoker = ClassCreator.builder().classOutput(classOutput).className(generatedName)
+                .interfaces(Invoker.class)
+                .build()) {
+
+            try (MethodCreator invoke = invoker.getMethodCreator(
+                    MethodDescriptor.ofMethod(generatedName, "invoke", Object.class, Object[].class))) {
+
+                // InjectableBean<Foo: bean = Arc.container().bean("1");
+                // InstanceHandle<Foo> handle = Arc.container().instance(bean);
+                // handle.get().ping();
+                ResultHandle containerHandle = invoke
+                        .invokeStaticMethod(MethodDescriptor.ofMethod(Arc.class, "container", ArcContainer.class));
+                ResultHandle beanHandle = invoke.invokeInterfaceMethod(
+                        MethodDescriptor.ofMethod(ArcContainer.class, "bean", InjectableBean.class, String.class),
+                        containerHandle, invoke.load(bean.getIdentifier()));
+                ResultHandle instanceHandle = invoke.invokeInterfaceMethod(
+                        MethodDescriptor.ofMethod(ArcContainer.class, "instance", InstanceHandle.class, InjectableBean.class),
+                        containerHandle, beanHandle);
+                ResultHandle beanInstanceHandle = invoke
+                        .invokeInterfaceMethod(MethodDescriptor.ofMethod(InstanceHandle.class, "get", Object.class),
+                                instanceHandle);
+
+                int parametersCount = method.parameters().size();
+                String[] argTypes = new String[parametersCount];
+                ResultHandle[] args = new ResultHandle[parametersCount];
+                for (int i = 0; i < parametersCount; i++) {
+                    // the only method argument of io.smallrye.reactive.messaging.Invoker is an object array so we need to pull out
+                    // each argument and put it in the target method arguments array
+                    args[i] = invoke.readArrayValue(invoke.getMethodParam(0), i);
+                    argTypes[i] = method.parameters().get(i).name().toString();
+                }
+                ResultHandle result = invoke.invokeVirtualMethod(
+                        MethodDescriptor.ofMethod(method.declaringClass().name().toString(), method.name(),
+                                method.returnType().name().toString(), argTypes),
+                        beanInstanceHandle, args);
+                if (VOID.equals(method.returnType().name())) {
+                    invoke.returnValue(invoke.loadNull());
+                } else {
+                    invoke.returnValue(result);
+                }
+            }
+        }
+
+        return generatedName.replace('/', '.');
+    }
+
+    private static Class<?> load(String className, ClassLoader cl) {
```

Comment:
that's sound like a very common method. Should it be moved to jandex?

Reply:
Very very common indeed. 
I don't know where exactly to place it since it's not really Jandex related.

Reply:
We can keep it for now.